--- Makefile	2026-06-24 19:41:33.472699014 +0800
+++ Makefile	2026-06-24 19:56:49.263089425 +0800
@@ -70,7 +70,8 @@ define Package/libunbound
     +PACKAGE_libunbound_libpthread:libpthread \
     +PACKAGE_libunbound_nghttp2:libnghttp2 \
     +PACKAGE_libunbound_pythonmodule:python3-base \
-    +PACKAGE_libunbound_dnstap:libprotobuf-c
+    +PACKAGE_libunbound_dnstap:libprotobuf-c \
+	+PACKAGE_libunbound_cachedb:libhiredis
 endef
 
 define Package/libunbound/description
@@ -155,6 +156,9 @@ define Package/libunbound/config
 	config PACKAGE_libunbound_dnstap
 		bool "Build with dnstap support."
 		default n
+	config PACKAGE_libunbound_cachedb
+		bool "Build with redis cache module support."
+		default n
 	endif
 endef
 
@@ -186,6 +190,7 @@ CONFIGURE_ARGS += \
     $(if $(CONFIG_PACKAGE_libunbound_python),--with-pythonmodule,) \
     $(if $(CONFIG_PACKAGE_libunbound_subnet),--enable-subnet,) \
     $(if $(CONFIG_PACKAGE_libunbound_dnstap),--enable-dnstap,) \
+	$(if $(CONFIG_PACKAGE_libunbound_cachedb),--enable-cachedb --with-libhiredis="$(STAGING_DIR)/usr",) \
 
 define Package/unbound-daemon/conffiles
 /etc/config/unbound
