--- Makefile
+++ Makefile
@@ -28,7 +28,7 @@ PKG_BUILD_PARALLEL:=1
 
 include $(INCLUDE_DIR)/package.mk
 
-ifeq ($(LIBNDPI_NDPIREADER),)
+ifeq ($(CONFIG_LIBNDPI_NDPIREADER),)
 CONFIGURE_ARGS += --with-only-libndpi
 endif
 
@@ -114,7 +114,7 @@ define Package/libndpi/install
 	$(CP) $(PKG_BUILD_DIR)/src/lib/libndpi.so* \
 		$(1)/usr/lib/
 	$(INSTALL_DIR) $(1)/usr/bin/
-ifneq ($(LIBNDPI_NDPIREADER),)
+ifneq ($(CONFIG_LIBNDPI_NDPIREADER),)
 	$(CP) $(PKG_BUILD_DIR)/example/ndpiReader \
 		$(1)/usr/bin/
 endif
