diff --git forkSrcPrefix/ndpi-netfilter/src/main.c forkDstPrefix/ndpi-netfilter/src/main.c
index 1463c642eaaa2a3630d499410450f3d9bdd983a7..aa77d70b851187a24a210e02c0d905b7a5344026 100644
--- forkSrcPrefix/ndpi-netfilter/src/main.c
+++ forkDstPrefix/ndpi-netfilter/src/main.c
@@ -3184,6 +3184,7 @@ static int __init ndpi_mt_init(void)
 	set_ndpi_malloc(malloc_wrapper);
 	set_ndpi_free(free_wrapper);
 
+#if 0
 	if(request_module("nf_conntrack") < 0) {
 		pr_err("xt_ndpi: nf_conntrack required!\n");
 		return -EOPNOTSUPP;
@@ -3198,6 +3199,7 @@ static int __init ndpi_mt_init(void)
 		return -EOPNOTSUPP;
 	}
 #endif
+#endif
 #ifdef NF_CT_CUSTOM
 	ret = nf_ct_extend_custom_register(&ndpi_extend,0x4e445049); /* "NDPI" in hex */
 	if(ret < 0) {
