diff --git forkSrcPrefix/ndpi-netfilter/src/main.c forkDstPrefix/ndpi-netfilter/src/main.c
index fbeabfb2d7fc38673283d2b839dc429b1b587033..f3f5d60f6264f07aaba28a86fb4c7a7e54a0b654 100644
--- forkSrcPrefix/ndpi-netfilter/src/main.c
+++ forkDstPrefix/ndpi-netfilter/src/main.c
@@ -3249,6 +3249,7 @@ static int __init ndpi_mt_init(void)
 	set_ndpi_malloc(malloc_wrapper);
 	set_ndpi_free(free_wrapper);
 
+#if 0
 	if(request_module("nf_conntrack") < 0) {
 		pr_err("xt_ndpi: nf_conntrack required!\n");
 		return -EOPNOTSUPP;
@@ -3263,6 +3264,7 @@ static int __init ndpi_mt_init(void)
 		return -EOPNOTSUPP;
 	}
 #endif
+#endif
 #ifdef NF_CT_CUSTOM
 	ret = nf_ct_extend_custom_register(&ndpi_extend,0x4e445049); /* "NDPI" in hex */
 	if(ret < 0) {
