diff --git forkSrcPrefix/ndpi-netfilter/src/main.c forkDstPrefix/ndpi-netfilter/src/main.c
index df404f646a431b288781361cca7182486a7322d5..bd41c120a860373e50c3f292b1de9c6d6fccf763 100644
--- forkSrcPrefix/ndpi-netfilter/src/main.c
+++ forkDstPrefix/ndpi-netfilter/src/main.c
@@ -3363,6 +3363,7 @@ static int __init ndpi_mt_init(void)
 	set_ndpi_malloc(malloc_wrapper);
 	set_ndpi_free(free_wrapper);
 
+#if 0
 	if(request_module("nf_conntrack") < 0) {
 		pr_err("xt_ndpi: nf_conntrack required!\n");
 		return -EOPNOTSUPP;
@@ -3377,6 +3378,7 @@ static int __init ndpi_mt_init(void)
 		return -EOPNOTSUPP;
 	}
 #endif
+#endif
 #ifdef NF_CT_CUSTOM
 	ret = nf_ct_extend_custom_register(&ndpi_extend,0x4e445049); /* "NDPI" in hex */
 	if(ret < 0) {
