diff --git forkSrcPrefix/ndpi-netfilter/ipt/libxt_ndpi.c forkDstPrefix/ndpi-netfilter/ipt/libxt_ndpi.c
index 758c48ab034cbd8eb53a91bb6e2171ca357e39ed..d497be8f7719376074a4852427fc67d5a4850022 100644
--- forkSrcPrefix/ndpi-netfilter/ipt/libxt_ndpi.c
+++ forkDstPrefix/ndpi-netfilter/ipt/libxt_ndpi.c
@@ -482,9 +482,21 @@ static int cmp_pname(const void *p1, const void *p2) {
 	return 0;
 }
 
+static char *strupr(char *s)
+{
+	char *tmp = s;
+
+	for (; *tmp; ++tmp) {
+		*tmp = toupper((unsigned char) *tmp);
+	}
+
+	return s;
+}
+
 static int ndpi_print_prot_list(int cond, char *msg) {
         int i,c,d,l,cp;
 	char line[128];
+	char name[32];
 	char *pn[NDPI_NUM_BITS+1];
 
 	bzero((char *)&pn[0],sizeof(pn));
@@ -506,6 +518,13 @@ static int ndpi_print_prot_list(int cond, char *msg) {
 		puts(msg);
 	qsort(&pn[0],NDPI_NUM_BITS,sizeof(pn[0]),cmp_pname);
 
+	for (i = 0; i <= NDPI_NUM_BITS; i++) {
+	    if(!pn[i]) break;
+	    strcpy(name, pn[i]);
+	    printf("--%s Match for %s protocol packets.\n", pn[i], strupr(name));
+	}
+
+#if 0
         for (i = 0,c = 0,l=0; i < NDPI_NUM_BITS; i++) {
 	    if(!pn[i]) break;
 	    l += snprintf(&line[l],sizeof(line)-1-l,"%-20s ", pn[i]);
@@ -517,6 +536,7 @@ static int ndpi_print_prot_list(int cond, char *msg) {
 	    }
 	}
 	if(c > 0) printf("%s\n",line);
+#endif
 	return d;
 }
 
