diff --git forkSrcPrefix/quickassist/build_system/build_files/env_files/environment.mk forkDstPrefix/quickassist/build_system/build_files/env_files/environment.mk
index 719611d55668d4223a71098260d14f4ed843e7ee..e020572248774a3b1e14ebc5ce2300747313f48b 100644
--- forkSrcPrefix/quickassist/build_system/build_files/env_files/environment.mk
+++ forkDstPrefix/quickassist/build_system/build_files/env_files/environment.mk
@@ -59,11 +59,7 @@ QAT_DIRECT_ROOT=$(ICP_ROOT)/quickassist/lookaside/access_layer/src/qat_direct/
 KERNEL_PATH=$(ICP_ROOT)/quickassist/lookaside/access_layer/src/qat_kernel/
 CMN_MEM_PATH=$(ICP_ROOT)/quickassist/utilities/libusdm_drv
 ICP_DEFENSES_ENABLED ?= y
-ifeq ($(shell cat /proc/kallsyms | grep -w "__stack_chk_fail" | wc -l), 1)
-KERNEL_DEFENSES_STACK_PROTECTION = y
-else
-KERNEL_DEFENSES_STACK_PROTECTION = n
-endif
+KERNEL_DEFENSES_STACK_PROTECTION ?= n
 
 #----------------------------------------------------------------------
 # ADF paths
