diff --git forkSrcPrefix/quickassist/utilities/osal/src/linux/user_space/OsalServices.c forkDstPrefix/quickassist/utilities/osal/src/linux/user_space/OsalServices.c
index 56c60714ee618fbc478f858adaffd3f543d0b832..1c83791b23ad0b0b28bd3749c041e614f3a9ee83 100644
--- forkSrcPrefix/quickassist/utilities/osal/src/linux/user_space/OsalServices.c
+++ forkDstPrefix/quickassist/utilities/osal/src/linux/user_space/OsalServices.c
@@ -360,6 +360,7 @@ OSAL_PUBLIC UINT64 osalTimestampGet(void)
 OSAL_PUBLIC UINT64 osalTimestampGetNs(void)
 {
     OsalTimeval ptime;
+#if defined(__GLIBC__) && defined(__GLIBC_PREREQ)
 #if __GLIBC_PREREQ(2, 17)
     struct timespec tspec;
 
@@ -372,6 +373,7 @@ OSAL_PUBLIC UINT64 osalTimestampGetNs(void)
             OSAL_LOG_DEV_STDOUT,
             "osalTimestampGetNs(): clock_gettime(CLOCK_REALTIME) system call "
             "failed. Invoking osalTimeGet() as fallback\n");
+#endif
 #endif
     ptime.secs = 0;
     ptime.nsecs = 0;
