diff --git forkSrcPrefix/configure.ac forkDstPrefix/configure.ac
index e3d17b90511741a567ff44bfd8a1db29a78bc51c..c35f0940bec2efa98eb5371628013b8bdb641f3b 100644
--- forkSrcPrefix/configure.ac
+++ forkDstPrefix/configure.ac
@@ -282,9 +282,6 @@ AC_SUBST(enable_qat_plock)
 #Default library name is qatengine
 AC_SUBST([LIBQATNAME], "qatengine")
 
-AC_CHECK_FILE(${qat_hw_dir_prefix}/include/qat/icp_sal_versions.h,
-	      [with_icp_sal_versions_h=yes],
-	      [with_icp_sal_versions_h=no])
 if test "x$with_icp_sal_versions_h" = "xyes"
 then
   if test `grep "define SAL_INFO2_DRIVER_SW_VERSION_TYPE \"in-tree\"" ${qat_hw_dir_prefix}/include/qat/icp_sal_versions.h | wc -l` = "1"
@@ -907,13 +904,13 @@ case "$host_os" in
 *linux*)
   if test "`cc --version | head -1 | awk '{print $3>=4.9}' 2>/dev/null`" = "1"
   then
-    AC_SUBST([cflags], ["-fPIC -Wall -Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fno-delete-null-pointer-checks -fwrapv -fstack-protector-strong -fno-strict-overflow"])
+    AC_SUBST([cflags], ["-fPIC -Wall -Wformat -Wformat-security -O2 -fno-delete-null-pointer-checks -fwrapv -fstack-protector-strong -fno-strict-overflow"])
   else
-    AC_SUBST([cflags], ["-fPIC -Wall -Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fno-delete-null-pointer-checks -fwrapv -fstack-protector -fno-strict-overflow"])
+    AC_SUBST([cflags], ["-fPIC -Wall -Wformat -Wformat-security -O2 -fno-delete-null-pointer-checks -fwrapv -fstack-protector -fno-strict-overflow"])
   fi
 ;;
 *freebsd*)
-    AC_SUBST([cflags], ["-fPIC -Wall -Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fno-delete-null-pointer-checks -fwrapv -fstack-protector-strong"])
+    AC_SUBST([cflags], ["-fPIC -Wall -Wformat -Wformat-security -O2 -fno-delete-null-pointer-checks -fwrapv -fstack-protector-strong"])
 ;;
 *) ;;
 esac
