diff --git forkSrcPrefix/ndpi-netfilter/src/Makefile forkDstPrefix/ndpi-netfilter/src/Makefile
index 357583a35c989bfe6c4bc166eeaac25fe6fd9ed4..9c0f01ccf2112bcb4cd57e89c8889f76da980ae3 100644
--- forkSrcPrefix/ndpi-netfilter/src/Makefile
+++ forkDstPrefix/ndpi-netfilter/src/Makefile
@@ -42,10 +42,10 @@ test::
 
 modules:
 	$(info $$xt_ndpi-y is [${xt_ndpi-y}] ${ccflags-y}) \
-	make  -C ${KERNEL_DIR} M=$$PWD $@ $(MFLAGS);
+	make ARCH=arm64 -C ${KERNEL_DIR} M=$$PWD $@ $(MFLAGS);
 
 modules_install:
-	make -C ${KERNEL_DIR} M=$$PWD $@;
+	make ARCH=arm64 -C ${KERNEL_DIR} M=$$PWD $@;
 	if [ "${WITHOUT_DEPMOD}" != "true" ]; then depmod -a; fi
 
 clean:
